src/org/rascalmpl/library/Set.rsc

```diff
@@ -513,3 +513,29 @@ public set[&T] union(set[set[&T]] sets) = {*s | s <- sets};
 
 @synopsis{Compute the Jaccard similarity between two sets.}
 real jaccard(set[value] x, set[value] y) = (1. * size(x & y)) / size(x + y);
+
+
+@synopsis{Calculate the intersection of a set of sets.}
+public set[&T] intersection(set[set[&T]] sets)  = (getFirstFrom(sets) | it & elem | elem <- sets);
```

Comment:
You could write this:
```
public set[&T] intersection({set[&T] firstSet, *set[&T] otherSets}) = (firstSet | it & elem | elem <- otherSets);
```

However, the function would then still throw an exception on the empty set (like the current version). If you want to avoid that add this alternative overload:

```
public set[&T] intersection({}) = {};
```